### DISCUSSION TITLE
What's the rule for ignoring types?

### DISCUSSION THREAD

**Discussion by lizongwu** [open-telemetry/opentelemetry-java-instrumentation#11489] [Q&A]
We're trying to instrument `RestTemplate.getForObject` method using `otel.instrumentation.methods.include=org.springframework.web.client.RestTemplate[getForObject]
`, but it doesn't seem to work—no span has been created for it.

And we noticed that many classes from Spring Framework are being ignored by the configuration in `AdditionalLibraryIgnoredTypesConfigurer`.

Could you clarify the rules for ignoring types and how to determine which types should be ignored?

**Comment by laurit** [comment]
> Could you clarify the rules for ignoring types and how to determine which types should be ignored?

I don't think there are any rules for this. The excludes try to limit the number of classes that are considered for instrumentation. I personally wish that we could just delete these.
